src/detect-filename.c

```diff
@@ -264,8 +264,12 @@ static uint8_t DetectEngineInspectFilename(DetectEngineCtx *de_ctx, DetectEngine
         if (buffer == NULL)
             continue;
 
+        if (buffer->inspect_offset > UINT32_MAX) {
```

Comment:
I really dislike these kind of changes, as it should have been limited before. This all adds runtime overhead.